tests/brevitas_examples/test_quantize_model.py

```diff
@@ -0,0 +1,722 @@
+            act_quant_type='sym',
+            scale_factor_type='float_scale',
+            quant_format='int',
+        )
+    # Test that negative values are invalid for bit widths
+    with pytest.raises(AssertionError):
+        quantize_model(
+            model=fx_model,
+            backend='fx',
+            weight_bit_width=-1,  # NOTE: invalid input
+            act_bit_width=-1,  # NOTE: invalid input
+            bias_bit_width=32,
+            weight_quant_granularity='per_tensor',
+            act_quant_percentile=99.9,
+            act_quant_type='sym',
+            scale_factor_type='float_scale',
+            quant_format='int',
+        )
+    # Test that invalid bias values are caught
+    with pytest.raises(KeyError):
+        quantize_model(
+            model=fx_model,
+            backend='fx',
+            weight_bit_width=8,
+            act_bit_width=8,
+            bias_bit_width=25,  # Valid values (16, 32)
+            weight_quant_granularity='per_tensor',
+            act_quant_percentile=99.9,
+            act_quant_type='sym',
+            scale_factor_type='float_scale',
+            quant_format='int',
+        )
+
+
+##########################
+# LAYERWISE MODE TESTING #
+##########################
+def test_layerwise_minifloat_invalid_bitwidths(minimal_model):
+    """
+    We test invalid mantissa and exponent bit widths. The mantissa + exponent + signed
+    should equal the total bit width for weights and activations, if doing
+    minifloat quantization (scale_factor_type='float_scale', quant_format='float').
+
+    We use a model with 3 input channels, and test `layerwise` quantization.
+    Because of the 3 input channels, this will set the quantization bit width
+    of the weights and activations of the first layer to be equal to
+    `layerwise_first_last_bit_width`.
+
+    We test:
+    - That the quantization will throw an error because the bit widths do not sum together correctly.
+    """
+
+    weight_bit_width = 8
+    act_bit_width = 8
+    bias_bit_width = 32
+
+    with pytest.raises(RuntimeError):
+        quantize_model(
+            model=deepcopy(minimal_model),
+            backend='layerwise',
+            weight_bit_width=weight_bit_width,
+            act_bit_width=act_bit_width,
+            bias_bit_width=bias_bit_width,
+            weight_quant_granularity='per_tensor',
+            act_quant_percentile=99.9,
+            act_quant_type='sym',  # signed bit = 1
+            scale_factor_type='float_scale',
+            quant_format='float',
+            layerwise_first_last_bit_width=13,  # invalid value, should be 11
+            layerwise_first_last_mantissa_bit_width=7,
+            layerwise_first_last_exponent_bit_width=3,
+            weight_mantissa_bit_width=6,
+            weight_exponent_bit_width=4,
+            act_mantissa_bit_width=5,
+            act_exponent_bit_width=5,
+        )
+
+
+def test_layerwise_valid_minifloat_bit_widths(minimal_model):
```

Comment:
This PR is getting there, but still need to work on this. I am testing if my explicit implementation of minifloat quantization matches the under-the-hood Brevitas one, but could use some guidance on whether this is correctly implemented. I'll be hacking at it either way!

Reply:
If it can be helpful, there are two PRs (https://github.com/Xilinx/brevitas/pull/922 and https://github.com/Xilinx/brevitas/pull/919) where we are expanding support to minifloat, to match the level of support we have for integer quantization. This means that a minifloat QuantTensor will have the correct metadata to properly  characterize it, and I think that could be helpful in the writing of the tests.

If you agree with this, I'm happy to leave the minifloat tests to another PRs after those two have been merged, as not to block this one.

Reply:
That sounds good!